cigaradvisor/blocks/search-results/search-worker.js

```diff
@@ -0,0 +1,156 @@
+/* eslint-disable max-len */
+/* eslint-disable no-restricted-globals */
+const ARTICLE_INDEX_PATH = '/cigaradvisor/index/article-index.json';
+const SEARCH_INDEX_PATH = '/cigaradvisor/index/search-index.json';
+const articleIndexData = [];
+
+const searchIndexData = [];
```

Comment:
I don't think this will ever be populated? Can you run some tests and see if this keeps data between calls?

Reply:
@bstopp you are correct...this is not getting cached between calls as we are fetching in worker... basically negating little perf improvement we could get by window.history.pushState and not reloading the page.

Reply:
What if the worker did the fetch and returned all the results. after that we don't have to actually run a search again per-page, we just change the page.

Reply:
@bstopp But I feel that would reset all the performance improvements we achieved...mapping entire list of search results with article-index was taking time. So, it would slow the search on first load.

Can we use Service Workers and Caching API ? Reading about them now...